src/Microsoft.Data.Analysis/PrimitiveColumnContainer.cs

```diff
@@ -14,15 +14,26 @@ namespace Microsoft.Data.Analysis
 {
     internal static class BitmapHelper
     {
+        private static ReadOnlySpan<byte> PopcountTable => new byte[] {
```

Comment:
Can you explain how this works/how it's being used?

Reply:
Sure. It's an array of bytes were each field shows haow many bits are set in an byte that is used as an index in this array. For example:
PopcountTable[0] is 0 // 00000000
PopcountTable[1] is 1 // 00000001
PopcountTable[2] is 1 // 00000010
PopcountTable[3] is 2 // 00000011
PopcountTable[4] is 1 // 00000100
PopcountTable[5] is 2 // 00000101
and etc

Actually is not my code, I took it  from Apache Arrow and modified. https://github.com/apache/arrow/blob/main/csharp/src/Apache.Arrow/BitUtility.cs
It's under Apache License, Version 2.0. Is there any problem with using code under this license in ML .Net?

Reply:
"It's under Apache License, Version 2.0. Is there any problem with using code under this license in ML .Net?"

@ericstj ?

Reply:
Can we just use their BitUtility? Do we already have a depedency?

Reply:
We can use for now, but there are issues with it. When I did a PR to Apache Arrow Eric Erhard asked me to create a task to make BitUtility internal. I am not sure that this task will be implemented in the nearest future, but still there is a possibility that this class will not be available externaly with one of the future Apache Arrow release. Another issue that we will not be able to change BitUtility. For example Apache version doesn't support long as an index, that I modified in current code

Reply:
If we copy code we should make sure to note it in https://github.com/dotnet/machinelearning/blob/main/THIRD-PARTY-NOTICES.TXT.

Reply:
Fixed

Reply:
@ericstj can you help review https://github.com/dotnet/machinelearning/pull/6846/files#diff-7d40364069786f7b06de93d93bd16adf4c2fb8e24a9b01be4f3ff0a2811d8798R88

Do we need to call out this file in particular or do anything different in this file?

Reply:
We follow the same requirements as dotnet/runtime - have a look at the history of https://github.com/dotnet/runtime/commits/main/THIRD-PARTY-NOTICES.TXT for other similar examples.